src/components/TextInputWithGlyph/index.tsx

```diff
@@ -4,14 +4,15 @@ import { size } from "constants/tokens";
 
 type TextInputWithGlyphProps = {
   icon: React.ReactElement;
+  inputRef?: React.RefObject<HTMLInputElement>;
 } & React.ComponentProps<typeof TextInput>;
 
 const TextInputWithGlyph: React.FC<TextInputWithGlyphProps> = (props) => {
```

Comment:
The right way is to use forwardRef when passing refs into functional components. 
<img width="1038" alt="image" src="https://user-images.githubusercontent.com/4605522/195440328-41657566-050b-4898-8b54-251689c1a4fa.png">